tests/test-pkcs11-tool-unwrap-wrap-test.sh

```diff
@@ -77,7 +77,66 @@ cmp plain_wrapped.key aes_plain_key >/dev/null 2>/dev/null
 assert $? "wrapped key after decipher does not match the original key"
 
 echo "======================================================="
-echo " RSA-PKCS Cleanup"
+echo " RSA-PKCS-OAEP Unwrap test"
+echo "======================================================="
+# RSA-PKCS-OAEP mechanism takes both a hash algorithm and MGF algorithm as parameters. For now we use SHA1, although it has been deprecated by NIST, because SoftHSM only supports SHA1 hash with OAEP currently. Known issue: https://github.com/softhsm/SoftHSMv2/issues/474 . When this issue is fixed, we shall replace with SHA256 or higher.
+# OpenSSL rsa_oaep_md hash algorithm option
+OSSL_OAEP_HASH_ALG="sha1"
+# pkcs11-tool hash-algorithm option
+P11_OAEP_HASH_ALG="SHA-1"
+# pkcs11-tool mgf option
+P11_OAEP_MGF_ALG="MGF1-SHA1"
+
+# Identifiers (pkcs11-tool --application-id argument) of PKCS11 key objects to be created
+ID4="97"
+ID5="98"
+
+# Reusing AES key and RSA key for unwrap/wrap operation already generated with ID1 in previous RSA-PKCS Unwrap test
+# Cleanup previously generated output
+rm aes_wrapped_key generic_extracted_key aes_ciphertext_openssl.data aes_ciphertext_pkcs11.data wrapped.key plain_wrapped.key
+
+# wrap AES key with RSA OAEP mode 
+openssl pkeyutl -encrypt -pkeyopt rsa_padding_mode:oaep -pkeyopt rsa_oaep_md:$OSSL_OAEP_HASH_ALG -pkeyopt rsa_mgf1_md:$OSSL_OAEP_HASH_ALG -pubin -keyform DER -inkey rsa_pub.key -in aes_plain_key -out aes_wrapped_key
```

Comment:
this command is failing with libressl, but the error message is not great:
```
  =======================================================
   RSA-PKCS-OAEP Unwrap test
  =======================================================
  -pkeyopt command before -inkey
  usage: pkeyutl [-asn1parse] [-certin] [-decrypt] [-derive] [-encrypt]
      [-hexdump] [-in file] [-inkey file] [-keyform fmt]
      [-out file] [-passin arg] [-peerform fmt]
      [-peerkey file] [-pkeyopt opt:value] [-pubin] [-rev]
      [-sigfile file] [-sign] [-verify] [-verifyrecover]
  ```
Sounds like the order of the arguments matter there?

Reply:
Ok should be fixed now.